pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -199,32 +212,174 @@ pub mod pallet {
+			ensure_root(origin)?;
+
+			Self::do_start_destroy_pool(pool_id, None)
 		}
 
-		// todo: check if we really need that tx.
 		#[pallet::call_index(8)]
 		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
-		pub fn destroy_accounts(_origin: OriginFor<T>) -> DispatchResult {
-			todo!()
+		pub fn refund_account(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			account: AccountIdLookupOf<T>,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+			let who = T::Lookup::lookup(account)?;
+
+			let pool_details = Pools::<T>::get(pool_id.clone()).ok_or(Error::<T>::PoolUnknown)?;
+
+			ensure!(pool_details.state.is_destroying(), Error::<T>::InUse);
+
+			let pool_account = pool_id.clone().into();
+
+			let mut total_collateral_issuance = Self::get_pool_collateral(&pool_account);
+
+			ensure!(!total_collateral_issuance.is_zero(), Error::<T>::ZeroAmount);
+
+			let total_issuances: Vec<(FungiblesAssetIdOf<T>, FungiblesBalanceOf<T>)> = pool_details
+				.bonded_currencies
+				.iter()
+				.map(|id| (id.clone(), T::Fungibles::total_issuance(id.clone())))
+				.filter(|(_, iss)| iss.gt(&Zero::zero()))
+				.collect();
+
+			let mut sum_of_issuances: FungiblesBalanceOf<T> = total_issuances
+				.iter()
+				.fold(Zero::zero(), |sum, (_, x)| sum.saturating_add(*x));
+
+			ensure!(!sum_of_issuances.is_zero(), Error::<T>::ZeroAmount);
+
+			let mut dead_currencies = Vec::with_capacity(total_issuances.len());
+			for (currency_id, token_issuance) in total_issuances.iter() {
+				let burnt = T::Fungibles::decrease_balance(
+					currency_id.clone(),
+					&who,
+					Bounded::max_value(),
+					Precision::BestEffort,
+					Preservation::Expendable,
+					Fortitude::Force,
+				)?;
+
+				let amount = burnt
+					.checked_mul(&total_collateral_issuance)
+					.ok_or(ArithmeticError::Overflow)? // TODO: do we need a fallback if this fails?
+					.checked_div(&sum_of_issuances)
+					.unwrap_or(Zero::zero());
+
+				T::CollateralCurrency::transfer(
+					T::CollateralAssetId::get(),
+					&pool_account,
+					&who,
+					amount,
+					Preservation::Expendable,
+				)?;
+
+				total_collateral_issuance -= amount;
+				sum_of_issuances -= burnt;
+
+				// if the total issuance drops to 0 due to this burn, kill the currency
+				if token_issuance <= &burnt {
+					dead_currencies.push(currency_id);
+				}
+			}
+
+			// destroy all active currencies if the collateral locked in the pool has dropped to 0
+			if total_collateral_issuance.is_zero() || sum_of_issuances.is_zero() {
+				// we assume all currencies that already had their total issuance at 0 have already been deactivated in a previous step
+				for (currency_id, _) in total_issuances {
+					T::Fungibles::start_destroy(currency_id, None)?;
+				}
+			} else {
+				// deactivate all currencies whose issuance dropped to 0 in this step
+				for currency_id in dead_currencies {
+					T::Fungibles::start_destroy(currency_id.clone(), None)?;
+				}
+			}
```

Comment:
In a previous version, I had an additional extrinsic to start_destroy currencies. This was clearly more efficient, but also made an already quite complex process even more complex.

Reply:
brought this back with d737d8a5237ad85eadd0e8787493334e0dbb89d7, where `start_destroy` is the only thing that calls `start_destroy` on the fungibles trait, and refunding is triggered instead with `start_refund`.